<ExceptionData>
  <Exceptions>
    <Locale name="English (United States)">
	<Exception name="Default">
		<Callbacks>
			<Callback className="mojo.km.exceptionhandling.PropagateExceptionCallback"/>
		</Callbacks>
	</Exception>
	
	<Exception name="mojo.km.messaging.exception.MessagingException">
        <Reason value="A notification exception has occurred."/>
        <Diagnosis value="A notification exception has occurred."/>
        <Solution value="Check the notification host server configuration"/>
		<Callbacks>
           <Callback className="mojo.km.notification.exception.NotificationExceptionCallback"/>
        </Callbacks>
    </Exception>
	
	<Exception name="java.lang.reflect.InvocationTargetException">
        <Reason value="A reflection exception has occurred."/>
        <Diagnosis value="A reflection exception has occurred."/>
        <Solution value="A reflection exception has occurred."/>
		<Callbacks>
           <Callback className="mojo.km.exceptionhandling.PropagateExceptionCallback"/>
        </Callbacks>
    </Exception>
	
	<Exception name="java.lang.ClassCastException">
        <Reason value="Invalid class cast."/>
        <Diagnosis value="Attempted to cast an object to a subclass of which it is not an instance."/>
        <Solution value="Check the string of the class name that the system is trying to load."/>
		<Callbacks>
           <Callback className="mojo.km.exceptionhandling.PropagateExceptionCallback"/>
        </Callbacks>
    </Exception>
	<Exception name="mojo.km.exceptionhandling.ParseRuntimeException">
        <Reason value="Error while parsing a date string."/>
        <Diagnosis value="Attempted to convert a string to a Date object, but that the string does not have the specified format."/>
        <Solution value="Check the specified format in the source code against what was entered."/>
		<Callbacks>
           <Callback className="mojo.km.exceptionhandling.StackTraceExceptionCallback"/>
        </Callbacks>
    </Exception>	
	<Exception name="mojo.km.messaging.exception.EventNotFoundException">
        <Reason value="The event service lookup failed."/>
        <Diagnosis value="Likely that the event class does not exist or the lookup is searching with the wrong key."/>
        <Solution value=""/>
		<Callbacks>
           <Callback className="mojo.km.exceptionhandling.PropagateExceptionCallback"/>
        </Callbacks>
      </Exception>	
	  <Exception name="java.lang.NumberFormatException">
        <Reason value="Error while parsing a number."/>
        <Diagnosis value="Attempted to convert a string to one of the numeric types, but that the string does not have the appropriate format."/>
        <Solution value="Check the string of the value that the system is trying to parse."/>
		<Callbacks>
           <Callback className="mojo.km.exceptionhandling.PropagateExceptionCallback"/>
        </Callbacks>
      </Exception>
	  <Exception name="java.lang.ClassNotFoundException">
        <Reason value="Tried to load in a class through its string name but no definition for the class with the specified name could be found."/>
        <Diagnosis value="Tried to load class through string name using one of the following: 1) The forName method in class Class. 2) The findSystemClass method in class ClassLoader. 3) The loadClass method in class ClassLoader."/>
        <Solution value="Check the string of the class name that the system is trying to load."/>
		<Callbacks>
           <Callback className="mojo.km.exceptionhandling.StackTraceExceptionCallback"/>
        </Callbacks>
      </Exception>
      <Exception name="java.lang.NullPointerException">
        <Reason value="Attempted to use an object that had a 'null' value."/>
        <Diagnosis value="One of the following has occurred: 1) Calling the instance method of a null object. 2) Accessing or modifying the field of a null object. 3) Taking the length of null as if it were an array. 4) Accessing or modifying the slots of null as if it were an array. 5) Throwing null as if it were a Throwable value."/>
        <Solution value="1) Identify object with a null value.  2) Check for null value before invoking method. OR Identify why value is getting set to null."/>
	  <Callbacks>
           <Callback className="mojo.km.exceptionhandling.PropagateExceptionCallback"/>
        </Callbacks>
      </Exception>
      <Exception name="edu.ucsb.ccs.jcontractor.InvariantViolationError">
        <Reason value="Class has an invalid state."/>
        <Diagnosis value="Contract for class require that each @invariant tag is met.  "/>
        <Solution value="Fix class to ensure a check for valid class state before executing any methods.  View call stack to determine location."/>
	  <Callbacks>
           <Callback className="mojo.km.exceptionhandling.StackTraceExceptionCallback"/>
        </Callbacks>
      </Exception>
      <Exception name="edu.ucsb.ccs.jcontractor.PreconditionViolationError">
        <Reason value="Method parameters have invalid states prior to execution of method."/>
        <Diagnosis value="Contract for methods require that each @pre tag is met.  "/>
        <Solution value="Fix method to ensure a check for valid parameter state before executing any methods.  View call stack to determine location."/>
	    <Callbacks>
           <Callback className="mojo.km.exceptionhandling.StackTraceExceptionCallback"/>
        </Callbacks>
      </Exception>
	  
	  <Exception name="mojo.km.context.multidatasource.ConnectionException">
        <Reason value="Error with datasource connection"/>
        <Diagnosis value="Resource adapter was unable to establish a connection.  "/>
        <Solution value="Check configuration parameters for connection"/>
	    <Callbacks>
           <Callback className="mojo.km.exceptionhandling.StackTraceExceptionCallback"/>
        </Callbacks>
      </Exception>
	  
      <Exception name="edu.ucsb.ccs.jcontractor.PostconditionViolationError">
        <Reason value="Method parameters/return has invalid state(s) after execution of method."/>
        <Diagnosis value="Contract for methods require that each @post tag is met.  "/>
        <Solution value="Fix method to ensure a check for valid parameter or return state post executing any methods.  View call stack to determine location."/>
	    <Callbacks>
           <Callback className="mojo.km.exceptionhandling.StackTraceExceptionCallback"/>
        </Callbacks>
      </Exception>

    <Exception name="Addin">
	  <Callbacks>
           <Callback className="com.togethersoft.modules.common.IdeExceptionCallbacks"/>
        </Callbacks>
      </Exception>
      <Exception name="mojo.km.context.TopLinkTransactionalContext.TransactionException">
        <Reason value="Error occured in TOPLink session"/>
        <Diagnosis value="Unable to perform TOPLink transaction."/>
        <Solution value="1) Check network transport to database.  2)  Check database instance still running."/>
    </Exception>

	<Exception name="mojo.km.context.multidatasource.M204Exception">
        <Reason value="Problem with the M204 Connection or Program Execution."/>
        <Diagnosis value="Make sure the subsystem is up and the 204 program runs with the data provided."/>
        <Solution value="Fix method to ensure a check for valid parameter. View call stack to determine location."/>
	  <Callbacks>
           <Callback className="mojo.km.exceptionhandling.PropagateExceptionCallback"/>
        </Callbacks>
     </Exception>
     <Exception name="mojo.km.context.multidatasource.VSAMException">
        <Reason value="Problem with the VSAM Connection or Program Execution."/>
        <Diagnosis value="Make sure CicsTransactionGateway and CICS are up."/>
        <Solution value="Fix method to ensure a check for valid parameter. View call stack to determine location."/>
	  	<Callbacks>
             <Callback className="mojo.km.exceptionhandling.PropagateExceptionCallback"/>
        </Callbacks>
      </Exception>
	  <Exception name="mojo.km.context.multidatasource.JDBCException">
        <Reason value="Problem with the JDBC Connection or Program Execution."/>
        <Diagnosis value="Contact JIMS2 Data Group to make sure DB2 is Available and SQL or Stored Procedure is Valid."/>
        <Solution value="Fix method to ensure a check for valid parameter. View call stack to determine location."/>
	  	<Callbacks>
           <Callback className="mojo.km.exceptionhandling.PropagateExceptionCallback"/>
        </Callbacks>
      </Exception>
    </Locale>
  </Exceptions>
</ExceptionData>
